### PR TITLE
Refactor `coordinates` tests to enable `black` for all the code

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

The primary goal of this pull request is to remove all `black` `# fmt: on/off` instructions from `coordinates` by refactoring the relevant code, but some additional minor refactoring was done in the affected files. In `coordinates` all the code ignored by `black` happened to be in the tests.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by mhvk — COMMENTED** [review]
Most changes are improvements generally, but one I thought was less good. I'd tend to keep that as is.

**Comment by pllim** [comment]
As a co-author of [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst) and also co-maintainer of coordinates, I feel like @eteq would have opinions on this.

**Comment by eerovaher** [comment]
> Most changes are improvements generally, but one I thought was less good. I'd tend to keep that as is.

@mhvk, did you forget to submit a comment pointing out which change you are talking about?

File: astropy/coordinates/tests/test_transformations.py
Comment (mhvk) [inline astropy/coordinates/tests/test_transformations.py:648]: Weird that my comment was lost, but this is the one that I felt was better as it was (though I would add a newline after `parametrize(` or move the opening bracket of the list to the first line.

It might also be possible to get things more on one line by using `ROT_30` etc instead of `ROTATION_30`.

Comment (eerovaher) [reply astropy/coordinates/tests/test_transformations.py:648]: Is it better now?

Comment (mhvk) [reply astropy/coordinates/tests/test_transformations.py:648]: Yes, quite a bit -- and good enough for me at least!

**Review by mhvk — APPROVED** [review]
This one seems to simply have fallen through the cracks, with me forgetting to explicitly approve after saying it all was OK. So, approving now, and merging.

**Comment by mhvk** [comment]
Actually, I cannot merge since CI is stuck. @eerovaher - maybe an idea to rebase while we are at it, just in case?

**Comment by pllim** [comment]
CI is green now. FYI.

**Pull request merged**
